typedb/stream/response_collector.py

```diff
@@ -46,46 +46,46 @@ def new_queue(self, request_id: UUID):
     def get(self, request_id: UUID) -> Optional["ResponseCollector.Queue[R]"]:
         return self._collectors.get(request_id)
 
-    def close(self, error: Optional[RpcError]):
+    def close(self, error: Optional[TypeDBClientException]):
         with self._collectors_lock:
             for collector in self._collectors.values():
                 collector.close(error)
 
-    def drain_errors(self) -> [RpcError]:
+    def get_errors(self) -> [TypeDBClientException]:
         errors = []
         with self._collectors_lock:
             for collector in self._collectors.values():
-                errors.extend(collector.drain_errors())
+                error = collector.get_error()
+                if error is not None:
+                    errors.append(error)
         return errors
 
-
     class Queue(Generic[R]):
 
         def __init__(self):
             self._response_queue: queue.Queue[Union[Response[R], Done]] = queue.Queue()
+            self._error: TypeDBClientException = None
 
         def get(self, block: bool) -> R:
             response = self._response_queue.get(block=block)
-            if response.message:
+            if response.is_response():
                 return response.message
-            elif response.error:
-                raise TypeDBClientException.of_rpc(response.error)
-            else:
+            elif response.is_done() and self._error is None:
                 raise TypeDBClientException.of(TRANSACTION_CLOSED)
+            elif response.is_done() and self._error is not None:
+                raise TypeDBClientException.of(TRANSACTION_CLOSED_WITH_ERRORS, self._error)
+            else:
+                raise TypeDBClientException.of(ILLEGAL_STATE)
```

Comment:
same as in client-nodejs: we now return the recorded error, and don't put the error in the `Done` queue slot